toucan_connectors/http_api/pagination_configs.py

```diff
@@ -0,0 +1,189 @@
+import logging
+from typing import Any, Optional
+from urllib.parse import parse_qs, urlparse
+
+from pydantic import BaseModel, Field
+
+from toucan_connectors.common import FilterSchema
+from toucan_connectors.http_api.http_api_data_souce import HttpAPIDataSource
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class PaginationConfig(BaseModel):
```

Comment:
Needs to inherit ABC as well to be considered abstract: 
* https://docs.python.org/3/library/abc.html 
* https://docs.pydantic.dev/latest/concepts/models/#abstract-base-classes